src/components/inspectors/SignalSelect.vue

```diff
@@ -0,0 +1,360 @@
+      this.signalName = signal.name;
+      this.showEditSignal = true;
+    },
+    getSignalById(id) {
+      return window.ProcessMaker.$modeler.definitions.rootElements.find(element => element.id === id);
+    },
+    change(value) {
+      if (!value) {
+        this.$emit('input', '');
+        this.fixUnrefreshedVueFormRender();
+        return;
+      }
+      let signal = this.getSignalById(value.id);
+      if (!signal) {
+        signal = window.ProcessMaker.$modeler.moddle.create('bpmn:Signal', {
+          id: value.id,
+          name: value.name,
+        });
+        window.ProcessMaker.$modeler.definitions.rootElements.push(signal);
+      }
+      this.$emit('input', this.storeId ? get(value, this.trackBy) : value);
+      this.fixUnrefreshedVueFormRender();
+    },
+    fixUnrefreshedVueFormRender() {
+      this.$nextTick(() => {
+        let form = this;
+        while (form && form.$options._componentTag !== 'vue-form-renderer') {
+          form = form.$parent;
+        }
+        if (form) {
+          form.$emit('update', form.transientData);
+        }
+      });
+    },
+    showAddSignal() {
+      this.showNewSignal = true;
+      this.signalId = '';
+      this.signalName = '';
+    },
+    cancelAddSignal() {
+      this.showNewSignal = false;
+      this.showEditSignal = false;
+    },
+    updateSignal() {
+      this.getSignalById(this.signalId).name = this.signalName;
+      this.showEditSignal = false;
+    },
+    addSignal() {
+      const signal = window.ProcessMaker.$modeler.moddle.create('bpmn:Signal', {
+        id: this.signalId,
+        name: this.signalName,
+      });
+      window.ProcessMaker.$modeler.definitions.rootElements.push(signal);
+      this.showNewSignal = false;
+    },
+    updateOptions() {
+      this.options = this.globalSignals;
+      window.ProcessMaker.$modeler.definitions.rootElements.forEach((element) => {
+        const localSignal = this.options.find(option => option.id === element.id);
+        if (element.$type === 'bpmn:Signal' &&!localSignal) {
+          this.options.push({
+            id: element.id,
+            name: element.name,
+          });
+        } else if (localSignal) {
+          localSignal.name = element.name;
+        }
+      });
+    },
+    loadOptions(filter) {
+      const pmql = this.pmql;
+      window.window.ProcessMaker.apiClient
+        .get(this.api, { params: { filter, pmql } })
+        .then(response => {
+          this.globalSignals = response.data.data || [];
+          this.updateOptions();
+        });
+    },
+    loadSelected(value) {
+      const signal = window.ProcessMaker.$modeler.definitions.rootElements.find(element => element.id === value);
+      if (signal) {
+        this.selectedOption = {
+          id: signal.id,
+          name: signal.name,
+        };
+      } else {
+        this.$emit('input', '');
+        this.fixUnrefreshedVueFormRender();
+      }
+    },
+    loadOptionsDebounced() {},
```

Comment:
This function is empty, can it be removed?

Reply:
removed

Reply:
No sorry , I am using that method in the <template>
And if I remove it I get an error/warning during the rendering:
```
Property or method "loadOptionsDebounced" is not defined on the instance but referenced during render.
```

Reply:
@caleeli Yes ... that's because you reference it a few lines above in the template; you should have removed it from there as well:
```
@search-change="loadOptionsDebounced"
```

Please go back and remove it from both places and see if you still have any errors.